From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sing-Han Chen <singhanc@nvidia.com>
Date: Mon, 27 May 2024 10:48:46 +0000
Subject: [PATCH 145/254] NVIDIA: SAUCE: usb: host: xhci-tegra: Add Tegra264
 XHCI support

This commit adds Tegra264 XUSB host mode controller support.

http://nvbugs/4295138

Change-Id: Ic7ee9b883f8e430abc000f7c770942afc02641ba
Signed-off-by: Sing-Han Chen <singhanc@nvidia.com>
(cherry picked from commit e968935c51e25c25e78c9b99e458c81352b9eac7)
Signed-off-by: Bodla Rakesh Babu <rbodla@nvidia.com>

diff --git a/drivers/usb/host/xhci-tegra.c b/drivers/usb/host/xhci-tegra.c
index 4caa62ba781ad..81fcc93080627 100644
--- a/drivers/usb/host/xhci-tegra.c
+++ b/drivers/usb/host/xhci-tegra.c
@@ -3515,6 +3515,33 @@ static const struct tegra_xusb_soc tegra234_vf4_soc = {
 	.lpm_support = true,
 };
 
+static const struct tegra_xusb_soc tegra264_soc = {
+	.supply_names = tegra194_supply_names,
+	.num_supplies = ARRAY_SIZE(tegra194_supply_names),
+	.phy_types = tegra194_phy_types,
+	.num_types = ARRAY_SIZE(tegra194_phy_types),
+	.max_num_wakes = 8,
+	.context = &tegra186_xusb_context,
+	.ports = {
+		.usb3 = { .offset = 0, .count = 4, },
+		.usb2 = { .offset = 4, .count = 4, },
+	},
+	.scale_ss_clock = false,
+	.has_ipfs = false,
+	.otg_reset_sspi = false,
+	.ops = &tegra234_ops,
+	.mbox = {
+		.cmd = XUSB_BAR2_ARU_MBOX_CMD,
+		.data_in = XUSB_BAR2_ARU_MBOX_DATA_IN,
+		.data_out = XUSB_BAR2_ARU_MBOX_DATA_OUT,
+		.owner = XUSB_BAR2_ARU_MBOX_OWNER,
+		.smi_intr = XUSB_BAR2_ARU_SMI_INTR,
+	},
+	.lpm_support = true,
+	.has_bar2 = true,
+	.has_pg_support = true,
+};
+
 static const struct of_device_id tegra_xusb_of_match[] = {
 	{ .compatible = "nvidia,tegra124-xusb", .data = &tegra124_soc },
 	{ .compatible = "nvidia,tegra210-xusb", .data = &tegra210_soc },
@@ -3525,6 +3552,7 @@ static const struct of_device_id tegra_xusb_of_match[] = {
 	{ .compatible = "nvidia,tegra234-xusb-vf2", .data = &tegra234_vf2_soc },
 	{ .compatible = "nvidia,tegra234-xusb-vf3", .data = &tegra234_vf3_soc },
 	{ .compatible = "nvidia,tegra234-xusb-vf4", .data = &tegra234_vf4_soc },
+	{ .compatible = "nvidia,tegra264-xusb", .data = &tegra264_soc },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, tegra_xusb_of_match);
-- 
2.43.0

